src/inc/sift-events/sift-events.php

```diff
@@ -564,102 +563,168 @@ public static function create_order( string $order_id, \WC_Order $order ) {
-		foreach ( $order->get_items( 'line_item' ) as $item ) {
-			if ( ! $item instanceof WC_Order_Item_Product ) {
-				// log an error...
-				wc_get_logger()->error( sprintf( 'Item not Item Product (order: %d).', $order->get_id() ) );
-				continue;
-			}
-			// Most of this we're basing off return value from `WC_Order_Item_Product::get_product()` as it will return the correct variation.
+		$browser = self::get_client_browser();
+		$ip      = $order->get_customer_ip_address() ?? self::get_client_ip();
+
+		$properties = array(
+			'$user_id'            => $user_id,
+			'$session_id'         => WC()->session?->get_customer_unique_id() ?? '',
+			'$order_id'           => $order_id,
+			'$user_email'         => $user_email,
+			'$verification_phone_number'
+				=> '+' === substr( $order->get_billing_phone(), 0, 1 ) ? preg_replace( '/[^0-9\+]/', '', $order->get_billing_phone() ) : null,
+			'$amount'             => self::get_transaction_micros( floatval( $order->get_total() ) ),
+			'$payment_methods'    => $sift_order->get_payment_methods(),
+			'$currency_code'      => $order->get_currency(),
+			'$billing_address'    => self::get_order_address( $order_id, 'billing' ),
+			'$shipping_address'   => self::get_order_address( $order_id, 'shipping' ),
+			'$expedited_shipping' => false,
+			'$items'              => array(),
+			'$browser'            => $browser,
+			'$ip'                 => $ip,
+			'$time'               => intval( 1000 * microtime( true ) ),
+		);
+
+		// Add the meta data.
+		foreach ( $order->get_items() as $item ) {
 			$product = $item->get_product();
-			if ( empty( $product ) ) {
-				// log an error...
-				wc_get_logger()->error( sprintf( 'Product not found for order %d.', $order->get_id() ) );
+
+			wc_get_logger()->debug(
+				$product,
+				array( 'source' => 'sift-free-orders-product' )
+			);
+
+			if ( ! ( $product instanceof WC_Product ) ) {
 				continue;
 			}
 
-			$items[] = array(
-				'$item_id'       => (string) $product->get_id(),
-				'$sku'           => $product->get_sku(),
+			$sku = empty( $product->get_sku() ) ? $product->get_id() : $product->get_sku();
+
+			// Construct item with ONLY fields specified in Sift API documentation.
+			$properties['$items'][] = array(
+				'$item_id'       => (string) $sku,
 				'$product_title' => $product->get_name(),
 				'$price'         => self::get_transaction_micros( floatval( $product->get_price() ) ),
-				'$currency_code' => $order->get_currency(), // For the order specifically, not the whole store.
-				'$quantity'      => $item->get_quantity(),
-				'$category'      => self::get_product_category( $product ),
-				'$tags'          => wp_list_pluck( get_the_terms( $product->get_id(), 'product_tag' ), 'name' ),
+				'$quantity'      => intval( $item->get_quantity() ),
+				'$currency_code' => $order->get_currency(),
+				'$category'      => self::get_product_category( $product ) ? self::get_product_category( $product ) : 'Uncategorized',
```

Comment:
Should we do it directly [here](https://github.com/woocommerce/sift-for-woocommerce/blob/trunk/src/inc/sift-events/sift-events.php#L1237) instead?

Reply:
Yeah, it needs to be updated; Sift usually kicks back when they expect a string and it's an empty string.